web-bluetooth/gap-characteristics.js

```diff
@@ -0,0 +1,107 @@
+function onButtonClick() {
+  log('Requesting Bluetooth Device...');
+  navigator.bluetooth.requestDevice(
+    {filters: [{name: ''}], optionalServices: ['generic_access']})
+  .then(device => {
+    log('Connecting to GATT Server...');
+    return device.gatt.connect();
+  })
+  .then(server => {
+    log('Getting GAP Service...');
+    return server.getPrimaryService('generic_access');
+  })
+  .then(service => {
+    log('Getting GAP Characteristics...');
+    return service.getCharacteristics();
+  })
+  .then(characteristics => {
+    let queue = Promise.resolve();
+    characteristics.forEach(characteristic => {
+      switch (characteristic.uuid) {
+
+        case BluetoothUUID.getCharacteristic('gap.appearance'):
+          queue.then(_ => readAppearanceValue(characteristic));
```

Comment:
Ah. Sorry for not noticing earlier. You need to do 
```js
queue = queue.then(...); 
```

Otherwise you are not queuing the promises. I've made all of these errors myself haha.